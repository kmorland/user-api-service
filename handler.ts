import { APIGatewayProxyEvent, APIGatewayProxyHandler, APIGatewayProxyResult, Callback, Context } from "aws-lambda";
import { ResponseService } from "./services/response.service";
import { UserService } from "./services/user.service";
// import { ValidateService } from "./services/validate.service";

import "source-map-support/register";

const userService: UserService = new UserService();

export const listUsers: APIGatewayProxyHandler = async (_event: APIGatewayProxyEvent, _context: Context, _callback: Callback<APIGatewayProxyResult>) => {
  _context.callbackWaitsForEmptyEventLoop = false;

  try {
    const { Items } = await userService.listUsers();

    return ResponseService.successResponse(Items);
  } catch (error) {
    return ResponseService.errorResponse(error);
  }
};
/*
export const createUser: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent, _context: Context, _callback: Callback<APIGatewayProxyResult>) => {
  _context.callbackWaitsForEmptyEventLoop = false;

  try {
    const { Item } = await userService.createUser(JSON.parse(event.body));

    return ResponseService.successResponse(Item);
  } catch (error) {
    return ResponseService.errorResponse(error);
  }
};

export const getUser: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent, _context: Context, _callback: Callback<APIGatewayProxyResult>) => {
  _context.callbackWaitsForEmptyEventLoop = false;

  if ( !ValidateService.isValidEmail(event.pathParameters.email) ) {
    return ResponseService.errorResponse( new Error("Email address required, invalid email address") );
  }

  try {
    const { Item } = await userService.getUser(event.pathParameters.email);

    return ResponseService.successResponse(Item);
  } catch (error) {
    return ResponseService.errorResponse(error);
  }
};

export const updateUser: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent, _context: Context, _callback: Callback<APIGatewayProxyResult>) => {
  _context.callbackWaitsForEmptyEventLoop = false;

  if ( !ValidateService.isValidEmail(event.pathParameters.email) ) {
    return ResponseService.errorResponse( new Error("Email address required, invalid email address") );
  }

  try {
    const { Item } = await userService.updateUser(event.pathParameters.email, JSON.parse(event.body));
    return ResponseService.successResponse(Item);
  } catch (error) {
    return ResponseService.errorResponse(error);
  }
};

export const deleteUser: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent, _context: Context, _callback: Callback<APIGatewayProxyResult>) => {
  _context.callbackWaitsForEmptyEventLoop = false;

  if ( !ValidateService.isValidEmail(event.pathParameters.email) ) {
    return ResponseService.errorResponse( new Error("Email address required, invalid email address") );
  }

  try {
    await userService.deleteUser(event.pathParameters.email);
    return ResponseService.successResponse();
  } catch (error) {
    return ResponseService.errorResponse(error);
  }
};
*/
