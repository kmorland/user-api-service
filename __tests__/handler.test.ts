// import { APIGatewayProxyEvent, Callback, Context } from "aws-lambda";
import { Context } from "aws-lambda";
import {} from "jest";
//import eventStub from "./stubs/list-user-event.json";
import { listUsers } from "../handler";
import * as AWS from "aws-sdk-mock";

// const mockEvent = {} as APIGatewayProxyEvent;
// const mockContext = {} as Context;
// const mockCallback = {} as Callback;

describe("User API Tests", () => {
    beforeAll(() => {
        AWS.mock("DynamoDB.DocumentClient", "scan", (params, callback) => {
            console.log(params);
            console.log(callback);
        });
    });
    
    test("listUsers should return HTTP Status 200", async () => {
        // const response = listUsers(mockEvent, mockContext, mockCallback);
        // expect(response.statusCode).toBe(200);
        //const event: APIGatewayProxyEvent = eventStub;
        const context = {};
        const result = await listUsers(null, context as Context, null);
        console.log("result", result);
        expect(result).toMatchSnapshot();
        expect(listUsers).toBeTruthy();
    });

    afterAll(() => {
        AWS.restore("DynamoDB.DocumentClient");
    });
});
