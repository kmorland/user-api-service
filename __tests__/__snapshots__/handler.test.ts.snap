// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User API Tests listUsers should return HTTP Status 200 1`] = `
Object {
  "body": "{\\"error\\":\\"Missing required key 'TableName' in params\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 400,
}
`;
